doc/source/Tutorials/Scripting/GettingStarted/index.md

```diff
@@ -1,152 +1,326 @@
+## "Hello, World!" and the _Script Editor_ ##
 
-The top field displays output from any commands executed, so we should see the greeting we printed along with the mysterious `script` variable that we printed with it. The `script` variable references the root node of a Gaffer file, and is our gateway to further exploration.
+Gaffer's _Script Editor_ lets you build and modify the node graph, edit and adjust the underlying code of your script (your project file), test expressions, and display results. By default, you can find the _Script Editor_ in the bottom-right panel, in a tab next to the _Scene Hierarchy_.
 
-> Tip : Small portions of the text input may be executed individually and kept around for later
-> editing by selecting them before hitting _Ctrl+Enter_. When there is no selection
-> _Ctrl+Enter_ executes everything and clears the contents of the input field, but the commands
-> can always be retrieved later by cutting and pasting them from the output field.
+![The Script Editor](images/scriptEditorEmpty.png "The Script Editor")
 
-Nodes and plugs
----------------
+The text field in the bottom-half is for inputting code, and behaves like a typical text editor: type, copy-paste, and enter new lines as you normally would. The top-half is a readout that displays the code you execute and any resulting output.
 
-The most important classes in the Gaffer API are the Nodes and Plugs which are connected together to make up a dependency graph. These are parented together in a hierarchy which nicely maps to python’s dictionary syntax. Plugs may be parented to nodes using Python's familiar dictionary notation :
+Try executing a "Hello, World!" command:
 
+1. Type `print "Hello, World!"` into the input field.
+
+2. Hit <kbd>Control</kbd> + <kbd>Enter</kbd> to execute the code.
+
+![The Script Editor with “Hello, World!”](images/scriptEditorHello.png "The Script Editor with “Hello, World!”")
+
+<!-- > Tip :-->
+<!-- > When you execute code, the contents of the input field will be erased. However, if you first highlight the code prior to executing it, the code you entered will remain in the field. -->
+
+
+## Scripting Basics with a Generic Node ##
+
+In this first section, we will cover the most basic operations for scripting nodes and plugs in Python, using a generic node as an example.
+
+
+### Creating a node ###
+
+Each node in the scripting API is a Python class.
+
+First, try creating a node of the generic node class:
+
+```python
+script.addChild( Gaffer.Node() )
 ```
-node = Gaffer.Node()
-node["firstPlug"] = Gaffer.IntPlug()
-node["secondPlug"] = Gaffer.FloatPlug()
+
+![Generic node in the Graph Editor](images/graphEditorGenericNode.png "Generic node in the Graph Editor")
+
+Notice that you added the node to the generic `script` variable.
+
+> Important :
+> The `script` variable is the root of, and primary context for, all Gaffer scripts. All nodes are added to it, and it contains all project-level variables.
```

Comment:
It's not clear to me if you're using "context" in the strict technical sense of `Gaffer.Context`, or a more general one. I'd suggest clarifying - perhaps this?

"""
The `script` is the root of node graph. All nodes are added to it, and it contains the project-level context variables that define the default execution context for the script.
"""

Although actually, I think it might be better to omit talk about contexts here, since it's a fairly complex topic, and not relevant to the rest of the tutorial.